R/engines.R

```diff
@@ -111,7 +111,7 @@ set_engine <- function(object, engine, ...) {
 set_engine.model_spec <- function(object, engine, ...) {
   mod_type <- class(object)[1]
 
-  if (rlang::is_missing(engine)) {
+  if (rlang::is_missing(engine) | is.null(engine)) {
```

Comment:
```suggestion
  if (rlang::is_missing(engine) || is.null(engine)) {
```

I believe this change will stop evaluating the expression once `rlang::is_missing(engine)` returns TRUE to avoid the `argument "engine" is missing, with no default` error from `is.null()` that's introduced in the snapshots below.